GameServerLib/Logic/GameObjects/Unit.cs

```diff
@@ -751,6 +758,46 @@ public ClassifyUnit ClassifyTarget(Unit target)
             return ClassifyUnit.Default;
         }
 
+        public byte GetNewBuffSlot(Buff b)
+        {
+            byte slot = GetBuffSlot(b, BuffSlotType.New);
+            AppliedBuffs[slot] = b;
+            return slot;
+        }
+
+        public void RemoveBuffSlot(Buff b)
+        {
+            byte slot = GetBuffSlot(b, BuffSlotType.Existing);
+            if (slot != 0x00) // Only remove the buff if it's already been applied
+            {
+                AppliedBuffs[slot] = null;
+            }
+        }
+
+        private byte GetBuffSlot(Buff b, BuffSlotType type)
```

Comment:
This function could still be simplified a bit 😄 

What I originally wished for when you merged your two separate functions into one was for them to share logic. Right now it's essentially two different functions inside the same body.

Think about the following example:
```C#
private byte GetBuffSlot(BuffSlotType type, Buff buff=null)
{
    for (byte i = 0; i < AppliedBuffs.Length; i++) // Find the first open slot
    {
        if (type == BuffSlotType.New && AppliedBuffs[i] == null)
        {
            return i; // If the slot is open and an open one was requested, return this slot
        }
        if (type == BuffSlotType.Existing && buff.Equals(b))
        {
            return i; // Returns the slot corresponding to the 
        }
    }
    throw new Exception("No slot found");
}
```
Less code used and cleaner, don't you agree?

This could be simplified even more if you scratch the `BuffSlotType` entirely, and instead use the buff parameter to decide what we're looking for, like so:

```C#
private byte GetBuffSlot(Buff buffToLookFor=null)
{
    for (byte i = 0; i < AppliedBuffs.Length; i++) // Find the first open slot
    {
        // I'm not sure if this matches buff.Equals(buffToLookFor) like you had originally,
        // so be sure to test for that
        if (AppliedBuffs[i] == buffToLookFor)
        {
            return i;
        }
    }
    throw new Exception("No slot found with requested value");
}
```

As always, let me know your thoughts 😄

Reply:
Hey, I like the first solution. I now understand what you meant back then. 🤔
But the second solution really is awesome. I love it! I'll check if ``==`` works but I'm pretty sure it will.
I find this default parameter value usage really smart, it makes the code super small and clean. 

Shall we still return ``0x00`` if we're looking for a Buff's slot, or shall we simply throw?
Thanks!

Reply:
I think I'd have the return type an `int` and return `-1` if nothing was found, thus allowing the error handling to be done elsewhere. No "right" choice here however, up to preference.

Reply:
Yeah seems good to me. I'll test it asap. By the way, IIRC, you can't have 0x00 as a buff slot, so I'll have to change that too. That's why I originally returned 0x00 when buff isn't applied

Reply:
If that's the case, 0x00 is alright too. Just as long as it's always the same and outside of the range of valid slots